### ISSUE TITLE
[native-image] static field changing during native image gen error message

### ISSUE THREAD

**Issue by aclement** [oracle/graal#1945]
Received whilst building the springmvc-tomcat sample (I can give you full repro instructions if useful).  Using graal dev build on commit #5a9edeb3ef07c (Java 11 base):
```
[tc:31247]        image:   3,235.14 ms
Fatal error: com.oracle.svm.core.util.VMError$HostedError: com.oracle.svm.core.util.UserError$UserException: Image heap writing found an object whose class was not seen as instantiated during static analysis. Did a static field or an object referenced from a static field change during native image generation? For example, a lazily initialized cache could have been initialized during image generation, in which case you need to force eager initialization of the cache before static analysis or reset the cache using a field value recomputation.
  object: com.oracle.svm.reflect.Class_getModule_68a51ac980c1dca447ffc0d0c4e92a101fd32949_3957@327f456b of class: com.oracle.svm.reflect.Class_getModule_68a51ac980c1dca447ffc0d0c4e92a101fd32949_3957
  reachable through:
    object: public java.lang.Module java.lang.Class.getModule()  of class: java.lang.reflect.Method
    object: public java.lang.Module java.lang.Class.getModule()  of class: java.lang.reflect.Method
    root: HostedField<Jre9Compat.getModuleMethod location: 27176   AnalysisField<Jre9Compat.getModuleMethod accessed: false reads: true written: true>>

	at com.oracle.svm.core.util.VMError.shouldNotReachHere(VMError.java:70)
	at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:648)
	at com.oracle.svm.hosted.NativeImageGenerator.lambda$run$0(NativeImageGenerator.java:452)
	at java.base/java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(ForkJoinTask.java:1407)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
```

**Comment by cstancu** [comment]
Yes, please provide repro instructions.

**Comment by cstancu** [comment]
I managed to get a minimal reproducible. The problem is caused by Tomcat's [`Jre9Compat.getModuleMethod`](https://github.com/apache/tomcat/blob/master/java/org/apache/tomcat/util/compat/Jre9Compat.java#L62). The repro is a boiled down version of that class:
```

import java.lang.reflect.Method;

public class Test {

    static class Jre9Compat {
        private static final Method getModuleMethod;
        static {
            Method m17 = null;
            try {
                m17 = Class.class.getMethod("getModule");
            } catch (ReflectiveOperationException | IllegalArgumentException e) {
                // Should never happen
                throw new Error("jre9Compat.unexpected", e);
            }
            getModuleMethod = m17;
        }

        public String getModuleName(Class<?> type) {
            try {
                Object module = getModuleMethod.invoke(type);
                return ((Module) module).getName();
            } catch (ReflectiveOperationException e) {
                return "no-module";
            }
        }
    }

    public static void main(String[] args) {
        Jre9Compat compat = new Jre9Compat();
        String moduleName = compat.getModuleName(String.class);
        if (!moduleName.equals("java.base")) {
            throw new Error("String should be in 'java.base' module.");
        }
        System.out.println("Success!");
    }
}
```

**Comment by cstancu** [comment]
Fixed by https://github.com/oracle/graal/commit/db5500192ea979c1ddcea6a851bc79f76c02e7cd. The fix will be ported to 19.3.1.
